File: apps/flutter_app/lib/models/seller_profile.dart
class SellerProfile {
  SellerProfile({
    required this.id,
    required this.userId,
    required this.storeName,
    required this.status,
  });

  final int id;
  final int userId;
  final String storeName;
  final String status;

  static SellerProfile fromJson(Map<String, dynamic> json) {
    return SellerProfile(
      id: json['id'] as int,
      userId: json['user_id'] as int,
      storeName: json['store_name'] as String,
      status: json['status'] as String,
    );
  }
}

File: backend/app/models/seller_profile.py
from __future__ import annotations

from datetime import datetime

from sqlalchemy import DateTime, ForeignKey, String, UniqueConstraint, func
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.core.db import Base


class SellerProfile(Base):
    __tablename__ = "seller_profiles"
    __table_args__ = (UniqueConstraint("user_id", name="uq_seller_profiles_user_id"),)

    id: Mapped[int] = mapped_column(primary_key=True)
    user_id: Mapped[int] = mapped_column(ForeignKey("users.id"), nullable=False)

    store_name: Mapped[str] = mapped_column(String(200))
    status: Mapped[str] = mapped_column(String(32), default="ACTIVE")

    created_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now()
    )
    updated_at: Mapped[datetime] = mapped_column(
        DateTime(timezone=True), server_default=func.now(), onupdate=func.now()
    )

    user = relationship("User", lazy="joined")

